### PR TITLE
Bugfix: Guideline with coordinate at zero is not selectable

### LINKED ISSUE #1525: Guideline with coordinate at zero is not selectable
In the `/E` glyph in MutatorSans.designspace, there is a guideline named "E bar", that has its x coordinate at zero. It is not displayed at the correct location, and is not selectable.

The default values for some Guideline fields are 0, and are therefor omitted from JSON dumps. When reconstructing a StaticGlyph from a JSON object (in its fromObject method), we must fill in any missing fields, so we have a "dense" object.

### PR DESCRIPTION
**Pull request by ollimeier**
Fixes #1525 

Fix guideline issue with normalizeGuidelines + update unittest

### CODE REVIEW

File: src/fontra/client/core/var-glyph.js [hunk 1]
```diff
@@ -58,9 +59,9 @@ export class StaticGlyph {
     source.components =
       (noCopy ? obj.components : obj.components?.map(copyComponent)) || [];
     source.anchors = noCopy ? obj.anchors || [] : copyCustomData(obj.anchors || []);
-    source.guidelines = noCopy
-      ? obj.guidelines || []
-      : copyCustomData(obj.guidelines || []);
+    source.guidelines = normalizeGuidelines(
```
[the remark this answers is not in the document]
Comment (ollimeier) [reply src/fontra/client/core/var-glyph.js]: Thanks. Changed it.

**Comment by ollimeier** [comment]
@justvanrossum May I ask you, if there is a reason for `locked: false` to be at the end of `return { ...identityGuideline, ...guideline, locked: false };` within `normalizeGuidelines`? 
This makes the guidelines always unlocked when I use `normalizeGuidelines`.
I am wondering if it would be better stored within `const identityGuideline = { x: 0, y: 0, angle: 0 };` -> `const identityGuideline = { x: 0, y: 0, angle: 0, locked: false };`

**Comment by justvanrossum** [comment]
> if there is a reason for locked: false to be at the end of return

Ahhh, hmmm. The reason is to ensure guidelines interpolate. But that is indeed not correct in this new case. Good catch!

Let me think what to do about it.

**Comment by justvanrossum** [comment]
> but it does not work for me and I don't know why 

&

> It does not work with UFO/Designspace

How does it not work? It seems to work for me.

File: src/fontra/client/core/utils.js [hunk 2]
```diff
@@ -546,9 +546,10 @@ export function areGuidelinesCompatible(parents) {
 
 const identityGuideline = { x: 0, y: 0, angle: 0 };
 
-export function normalizeGuidelines(guidelines) {
+export function normalizeGuidelines(guidelines, resetLocked = false) {
   return guidelines.map((guideline) => {
-    return { ...identityGuideline, ...guideline, locked: false };
+    const guidelineLocked = resetLocked ? false : guideline.locked ? true : false;
```
Comment (justvanrossum) [inline src/fontra/client/core/utils.js:551]:
```suggestion
    const guidelineLocked = resetLocked ? false : !!guideline.locked;
```
And then you might as well inline the expression in the line below again.

**Comment by ollimeier** [comment]
> > but it does not work for me and I don't know why
> 
> &
> 
> > It does not work with UFO/Designspace
> 
> How does it not work? It seems to work for me.

I was very unclear. I am sorry for that. 
The mentioned bug has been fixed. But I noticed, that when you lock a guideline (UFO/Designspace) and do a refresh, the guideline is no longer locked. So, it does not save the 'lock'-status.

**Pull request merged**